### PR TITLE
Cmake policy documentation and buiding instructions

### PR DESCRIPTION
**Pull request by mBialczak**
Adding 2 documentation items:
1. cmake-policy dokumentation
2. building instructions

### CODE REVIEW

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: documentation/cmake_policy/Cmake_policy.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+![](image-snippets/image-5.png)
+
+The purpose of this file is to separate potentially ugly installation rules for external libraries, which we want to keep fixed and common for all devs to use as common environment as possible (in some cases it might even be necessary)
+
+For starter it contains one rule for `protobuf` library:
+
+![](image-snippets/image-8.png)
+
+Not all in this file is obvious for me. I trusted GH Copilot with that and only adjusted to work.
+I consider it to be experimental for the time being.
```
[the remark this answers is not in the document]
Comment (mBialczak) [reply documentation/cmake_policy/Cmake_policy.md]: done

File: documentation/building_and_running_tests.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+## Building backend and tests manually
+
+If we want to build `backend` modules from clean state, we need to:
+1. Create or cleanup "build" directory
+2. ```cd build```
+3. Run cmake configuration:
+   ```cmake -DCMAKE_BUILD_TYPE=Debug ../modules/backend/```
+4. Build `backend` module
+   ```cmake --build . --parallel --target install -- -j2```
+   NOTE: `-j2` flag sets maximum number of threads used. It origins in attempt to fix building on some machines.
```
[the remark this answers is not in the document]
Comment (mBialczak) [reply documentation/building_and_running_tests.md:15]: done

File: documentation/cmake_policy/Cmake_policy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mBialczak) [reply documentation/cmake_policy/Cmake_policy.md]: Thanks, removed.

File: documentation/cmake_policy/Cmake_policy.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+Coming back to main CMakeLists.txt, we find try to find the version matching the one we requested in helper `installation_rules.cmake` file:
+
+![](image-snippets/image-9.png)
+
+Hopefully, after running cmake all above will install and find the specific, fixed version placed in external_libraries/protobuf folder.
```
[the remark this answers is not in the document]
Comment (mBialczak) [reply documentation/cmake_policy/Cmake_policy.md]: Removed, thanks

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Pull request merged**